client/contract.go

```diff
@@ -31,35 +32,37 @@ func NewContractClient(endpointURL string) (*ContractClient, error) {
 	}, nil
 }
 
-// ContractInfo returns the ContractInfo for a specific address
-func (c *ContractClient) ContractInfo(contractAddress common.Address, isErc20 bool) (*ContractInfo, error) {
-	cachedInfo, isCached := c.cache.Get(contractAddress)
-
-	if isCached {
-		castCachedInfo := cachedInfo.(*ContractInfo)
-		return castCachedInfo, nil
+// ContractCurrency returns the currency for a specific address
+func (c *ContractClient) ContractCurrency(addr common.Address, erc20 bool) (*types.Currency, error) {
+	if currency, cached := c.cache.Get(addr); cached {
+		return currency.(*types.Currency), nil
 	}
 
-	token, err := NewContractInfoToken(contractAddress, c.ethClient)
+	token, err := NewContractInfoToken(addr, c.ethClient)
 	if err != nil {
 		return nil, err
 	}
+
 	symbol, symbolErr := token.Symbol(nil)
 	decimals, decimalErr := token.Decimals(nil)
 
 	// Any of these indicate a failure to get complete information from contract
 	if symbolErr != nil || decimalErr != nil || symbol == "" || decimals == 0 {
-		if isErc20 {
+		if erc20 {
 			symbol = UnknownERC20Symbol
 			decimals = UnknownERC20Decimals
 		} else {
 			symbol = UnknownERC721Symbol
 			decimals = UnknownERC721Decimals
 		}
 	}
-	contractInfo := &ContractInfo{Symbol: symbol, Decimals: decimals}
+
+	currency := &types.Currency{
```

Comment:
I don't like the partial population of this. We should use the existing code (with a different var name)

Reply:
resolved in 1c42325